### DISCUSSION TITLE
Ortur Laser Master 2  - can't change laser power

### DISCUSSION THREAD

**Discussion by choucroutemelba** [arkypita/LaserGRBL#1812] [General]
Hello to the comunity.

I noticed that the laser power is always high even when framing.
Ortur laser master 2 - 1.42 firmware

During framing the laser is still engraving even with S changed from 5/255 to 1/255 power.

Do you have any idea to solve this issue ?

Here the button parameter :
G0 X[left] Y[bottom]
M3 S1 F1000 
G1 Y[top] 
G1 X[right]
G1 Y[bottom] 
G1 X[left]
M5
G0 X0 Y0

**Comment by Tribo-2dot0** [comment]
Try changing $30=1000 and you will have smaller fractions to work with the laser power.

**Comment by choucroutemelba** [comment]
Thks for your feedback

II already check this point... same situation.
I'll investigate the PWM signal send from the motherboard to the laser.
At least it'll help to localize the issue.

**Comment by StuartB4** [comment]
I would check where the laser plugs in to the little circuit board. There have been a lot of these boards that have had 
bad soldering on the pins, leading to dry joints/bad connections. 
![image](https://user-images.githubusercontent.com/16980026/163052521-daaba87a-3ba1-4f62-a817-00f9031752b9.png)

**Reply by choucroutemelba** [reply]
Thks for this clue...
I'll start with this check !

**Comment by choucroutemelba** [comment]
Hello
I made some mesurements.

With M3 S30 F1000 I got a 118µs positive pulse.

30/255 = 11,7 %
That means  the cycle is about 1ms long (118µs/0.117) => 1000hz PWM signal

The pulse reacts according to the laserGRBL command witch is the expected behaviour.
Nevertheless, my laser is still on high power and able to engrave plywood at 0% off power.

![20220414_111236 1](https://user-images.githubusercontent.com/34708200/163357880-e9dc1c83-2c03-4020-8271-5747ef565f61.jpg)

**Comment by choucroutemelba** [comment]
Hello

I'll focus on the laser driver PCB called :LY-BC-0160
A found this interesting link below :
[https://endurancelasers.com/testing-of-an-ortur-diode-blue-5500-mw-5-5-watt-laser-module/](url)

here the electronic circuit :
![20220414_112148 1](https://user-images.githubusercontent.com/34708200/163361045-df226235-52c0-4d4d-aee9-ae1e9c25b6ab.jpg)

**Reply by StuartB4** [reply]
Link seems to be incorrect, it has 2 address's.
It goes to here. (https://github.com/arkypita/LaserGRBL/discussions/url)

[https://endurancelasers.com/testing-of-an-ortur-diode-blue-5500-mw-5-5-watt-laser-module](https://github.com/arkypita/LaserGRBL/discussions/url)

It should be just this. 
https://endurancelasers.com/testing-of-an-ortur-diode-blue-5500-mw-5-5-watt-laser-module

**Comment by choucroutemelba** [comment]
Thks Stuart !

I made some additionnal tests.
By raising the voltage above 1.6v the laser turns on
By reducing the voltage at 1.4 V the laser turns off.

That's not consistent with the datasheet (see  below). Or am I wrong ?

Nevertheless, at the scope we can see that low level is about 1.8V which is above 1.4v.
The laser can't see the low level that's why he runs always at full power.
![dimming](https://user-images.githubusercontent.com/34708200/163404044-1d951f13-6f13-4383-b385-ab7642de68a8.PNG)
![dimming_values](https://user-images.githubusercontent.com/34708200/163404066-7d51e1dc-ef21-4835-8f9f-9a67afffc727.PNG)

**Comment by choucroutemelba** [comment]
What's your experience ?

**Comment by arkypita** [comment]
Hi 
I had OLM2 20W laser and other laser modules based on LY-BC-0160 driver board (like older atomstack A5 and all works fine. I think you have a faulty module.

**Comment by arkypita** [comment]
```
By raising the voltage above 1.6v the laser turns on
By reducing the voltage at 1.4 V the laser turns off.

That's not consistent with the datasheet (see below). Or am I wrong ?
```

Datasheet values 0.6 and 2.8 are max and min level, 

I think that they tell you that under 0.6 is low lever for sure and over 2.8 are high level for sure but not that this turn off at 0.6 and on at 2.8 exactly.

**Comment by choucroutemelba** [comment]
Thks for the clarification

What is not clear to me is the lowest high level and highest low level to check if my signal is inside the limits. 

What is somehow inconsistent is that my low level PWM signal (1.8V) coming from the ECU is above the mesured turn off low level (1.4v).
So one the laser has been powered on it never stops.

I asked my friend to make some mesurements to be able to define who is faulty.

**Reply by arkypita** [reply]
With **unconnected** laser module the PWM signal coming from control board should be 0-5v or 0-3.3v (based on board logic).
If not, your controller is faulty.

**Comment by choucroutemelba** [comment]
Buying a new controller is pretty expensive (approx. 70€) + 2 month delay.
Do you know where I can find the schematic of the OLM2 controller ? 
Maybe I can fix it myself ...

**Comment by StuartB4** [comment]
If you join the official Ortur Laser Group on Facebook and search for Gil Araújo, he is Ortur's Customer Support guy. 
Depending on how old your machine is he may be able to help, he's really good from what I've seen at sorting problems out.

https://www.facebook.com/groups/orturusergroup

![image](https://user-images.githubusercontent.com/16980026/164896964-136ccc3b-94bb-499b-b661-e128333977c4.png)

**Comment by arkypita** [comment]
unfortunately there are no public schematics of hardly any markers, but the circuitry is quite simple, just follow the PWM pin back to the microcontroller. You shouldn't find many components, maybe none!

Anyway I suggest you do a firmware flash and a reset of the board configuration. Maybe this is enough to solve your problem

**Comment by choucroutemelba** [comment]
Thks for your support.

I've just ordered a new controller

**Comment by choucroutemelba** [comment]
I've just got a new controller.
Indeed it solves the issue.
On the picture below the pulse is goes from 0v to 2.5V.

If I have a couple of hours to spend I'll try to fix the faulty controller.

Thks for your support.
![20220505_141957](https://user-images.githubusercontent.com/34708200/166924393-6e3e9bef-0b3a-4482-a30a-c65efe47aba7.jpg)

**Comment by arkypita** [comment]
Glad to know the problem is solved, thanks for giving us feedback.
If you can fix the  faulty controller write us as it surely could be useful for others.

**Comment by choucroutemelba** [comment]
Sure.

With a quick check on the PCB  I found out a very basic circuitry.

=> µc + pull up resistor  + filtration cap + Led indicator (+resistor).

My assumption was a faulty output pin on the µc (due to whatever reason).

But indeed it would be helpfull for all the share full schematic.
